### PR TITLE
Process.GetProcessById bugfix

### PR DESCRIPTION
**Pull request by epeshk**
https://github.com/dotnet/runtime/issues/63937

### CODE REVIEW

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs [hunk 1]
```diff
@@ -28,15 +28,16 @@ public static bool IsProcessRunning(int processId)
         public static bool IsProcessRunning(int processId, string machineName)
         {
             // Performance optimization for the local machine:
-            // First try to OpenProcess by id, if valid handle is returned, the process is definitely running
+            // First try to OpenProcess by id, if valid handle is returned verify that process is running
             // Otherwise enumerate all processes and compare ids
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs:32]: @adamsitnik, do you know why we fall back to the IndexOf here for the !IsRemoteMachine case?  I'm wondering why the using block below can't just contain `return !processHandle.IsInvalid && !HasExited(processHandle, ref signaled, out _);`.

**Comment by stephentoub** [comment]
Thanks for working on this. Can you please add a test that would have failed before and will now succeed?

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs (hunk 1 above)
Comment (epeshk) [reply src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs:32]: https://github.com/dotnet/runtime/pull/65041

**Comment by epeshk** [comment]
I've added test for this

**Review by jozkee — APPROVED** [review]

**Pull request merged**